### PR TITLE
Fixing re-render issue for empty vertical bar charts

### PR DESCRIPTION
**Pull request by srmukher**
## Previous Behavior

Work item https://uifabric.visualstudio.com/iss/_workitems/edit/8408/
Currently, once the vertical bar chart is mounted as empty, it is not re-rendering even if it has some data later when the component is updated.

## New Behavior

ComponentDidUpdate life cycle method now handles the issue when the empty chart is updated. Also added tests to verify the re-rendering scenario using react testing library.

## Related Issue(s)

- Fixes https://uifabric.visualstudio.com/iss/_workitems/edit/8408/

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/711a734d30fab93fd8359e2880329beacd237e2a'>711a734d30fab93fd8359e2880329beacd237e2a</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 1]
```diff
@@ -106,7 +106,15 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
 
   public componentDidMount(): void {
     const isChartEmpty =
-      this.state.emptyChart ||
+      this._points.length === 0 ||
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:109]: Is this issue only for vertical bar charts and not others?

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { VerticalBarChart } from './VerticalBarChart';
+
+describe('Vertical bar chart rendering', () => {
+  test('Should re-render the vertical bar chart with data', async () => {
+    // Arrange
+    const { container, rerender } = render(<VerticalBarChart data={[]} />);
+    const getById = queryAllByAttribute.bind(null, 'id');
+    // Assert
+    expect(container).toMatchSnapshot();
+    screen.debug(container, 500000);
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx:13]: is screen.debug call intended.

Comment (srmukher) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChartRTL.test.tsx:13]: removed

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx (hunk 1 above)
Comment (srmukher) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:109]: as discussed offline, raising a separate PR for all other charts

File: packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx [hunk 3]
```diff
@@ -106,7 +106,15 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
...
+      (d3Max(this._points, (point: IVerticalBarChartDataPoint) => point.y)! <= 0 && !this._isHavingLine);
+    if (this.state.emptyChart !== isChartEmpty) {
+      this.setState({ emptyChart: isChartEmpty });
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx:112]: Could you clarify that setting it as state variable is preferable than checking at render time.

Comment (srmukher) [reply packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx]: Removing the state reduces number of re-renders, so updated it accordingly.

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**